app/web_hub.go

```diff
@@ -250,85 +292,82 @@ func (a *App) invalidateCacheForUserTeams(userId string) {
 	}
 }
 
-func (a *App) invalidateCacheForUserSkipClusterSend(userId string) {
-	a.Srv().Store.Channel().InvalidateAllChannelMembersForUser(userId)
-
-	hub := a.GetHubForUserId(userId)
-	if hub != nil {
-		hub.InvalidateUser(userId)
-	}
-}
-
-func (a *App) invalidateCacheForUserTeamsSkipClusterSend(userId string) {
-	hub := a.GetHubForUserId(userId)
-	if hub != nil {
-		hub.InvalidateUser(userId)
-	}
-}
-
-func (a *App) invalidateCacheForWebhook(webhookId string) {
-	a.Srv().Store.Webhook().InvalidateWebhookCache(webhookId)
-}
-
-func (a *App) InvalidateWebConnSessionCacheForUser(userId string) {
-	hub := a.GetHubForUserId(userId)
-	if hub != nil {
-		hub.InvalidateUser(userId)
-	}
-}
-
+// UpdateWebConnUserActivity sets the LastUserActivityAt of the hub for the given session.
 func (a *App) UpdateWebConnUserActivity(session model.Session, activityAt int64) {
 	hub := a.GetHubForUserId(session.UserId)
 	if hub != nil {
 		hub.UpdateActivity(session.UserId, session.Token, activityAt)
 	}
 }
 
+// Register registers a connection to the hub.
 func (h *Hub) Register(webConn *WebConn) {
 	select {
 	case h.register <- webConn:
-	case <-h.didStop:
+	case <-h.stop:
 	}
 }
 
+// Unregister unregisters a connection from the hub.
 func (h *Hub) Unregister(webConn *WebConn) {
 	select {
 	case h.unregister <- webConn:
 	case <-h.stop:
 	}
 }
 
+// Broadcast broadcasts the message to all connections in the hub.
 func (h *Hub) Broadcast(message *model.WebSocketEvent) {
 	if h != nil && h.broadcast != nil && message != nil {
```

Comment:
Will check .. and if not needed, will remove.

Reply:
done

Reply:
Well .. surprise surprise. This was needed after all :)


The hub nil check is required because of the wonderful way we setup our tests :) We call `s, err := app.NewServer(options...)` which returns a server, but only after that, we call `wsapi.Init(th.App, th.App.Srv().WebSocketRouter)` through our FakeApp adapter to initialize the hub. But in the `NewServer` call itself, we call `RunOldAppInitialization` which directly proceeds to broadcast some messages happily. 

```
message.Add("config", s.FakeApp().ClientConfigWithComputed())
s.Go(func() {
	s.FakeApp().Publish(message)
})
```

Therefore, to make this work, we incur a nil check for every single message broadcast. Just found this out because CI failed on hub Broadcast crash.